mteb/abstasks/AbsTaskZeroshotClassification.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import tqdm
+from datasets import Dataset
+
+from mteb.encoder_interface import Encoder, EncoderWithQueryCorpusEncode
+from mteb.load_results.mteb_results import ScoresDict
+
+from ..evaluation.evaluators import ZeroshotClassificationEvaluator
+from .AbsTask import AbsTask
+
+logger = logging.getLogger(__name__)
+
+
+class AbsTaskZeroshotClassification(AbsTask):
+    """Abstract class for ZeroshotClassification tasks
+    The similarity between an images and candidate text prompts, such as this is a dog/this is a cat.
+
+    self.load_data() must generate a huggingface dataset with a split matching self.metadata_dict["eval_splits"], and assign it to self.dataset. It must contain the following columns:
+        png: list of Image.Image
+        cls: list of int
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def _add_main_score(self, scores) -> None:
+        scores["main_score"] = scores[self.metadata.main_score]
+
+    def _evaluate_subset(
+        self,
+        model: EncoderWithQueryCorpusEncode | Encoder,
```

Comment:
Well it isn't quite an Encoder (as the encoder is defined now), maybe we should add an ImageEncoder (and ideally make Encoder -> TextEncoder)? Do we need some way to differentiate between the two?

Reply:
Yep not sure what the best approach is for this. 

We could have 
```
Encoder # for the others to subclass from
TextEncoder # the encoder atm.
ImageEncoder 
ImageTextEncoder # the multimodal version can take both images and text
```

Reply:
Yeah, will work on this with `mieb.get_model() ` after done with tasks. The `Encoder` thing was more of a placeholder when following previous tasks template! :)